### DISCUSSION TITLE
Dual Battery support for the battery widget on Qtile (battery merging/fusion)

### DISCUSSION THREAD

**Discussion by Sniki** [qtile/qtile#4143] [Ideas]
Hi guys,

I have a Lenovo Thinkpad X240 which does have two batteries, i run Qtile on all my devices (Laptops and Desktops).
widget.battery is just perfect for single battery devices however it would really be helpful to add a feature to it to support two batteries, what i mean support for two batteries is merged calculation from both of them:
- Report total battery percentage left from both batteries (calculated based on their design/total capacity), as the capacity is different from the internal and external battery
- Report total time left to charge the battery (calculated approximately for both batteries) and also report total time left on battery (calculated from both batteries.

So basically to have a fusion feature so the widget reports as if the two batteries were one/merged.

Right now i have two battery widgets for each one but it takes space on the bar and kinda confusing/messy and i would rather have another widget instead of the second battery widget.
Qtile Extras does have a widget that displays both batteries at one widget but again, it shows two battery icons and the hole point is that i wouldn't care how much charge does each one have but i just need to know how much percentage battery have i left on my laptop like it had only a single one.

I know that it's possible to make a script and use it  for what i need but it needs some math and i am not good at scripting or better say newbie, i can just edit some simple stuff from a script but not able to write on my own.

I found something as a base from a guy there for the exact same laptop but it has separate scripts for that and can serve as a base for it: https://github.com/pcchou/dual-battery-thinkpad-upower-script

It would be nice to have those merged (i don't know the information about how much watt is it consuming/charging).

Thanks in advance !
Sorry for the somewhat TLDR

**Comment by fjpavon** [comment]
Hi Sniki, 
It seems a bit old the message but i have the same "issue" (thinkpad T470s). It doesn't seems to be difficult to improve the code and include all the batteries so I'm going to modify the widget. 
I think that the most difficult task for me it's going to be work with git....

**Comment by fjpavon** [comment]
Hi
I have end the implementation. It's working with linux and FreeBSD. I never contributed to any project so the steps to be done, like create the pr, etc, are unknown for me...

**Reply by tych0** [reply]
there is a bit of a walkthrough here: https://docs.qtile.org/en/stable/manual/contributing.html#writing-code

**Reply by fjpavon** [reply]
For sure. I have some questions about what i get launching pytest and tox but i'll open another thread.
